app/serializers/activitypub/context_serializer.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class ActivityPub::ContextSerializer < ActivityPub::Serializer
+  include RoutingHelper
+
+  attributes :id, :type, :first, :attributed_to
+
+  has_one :first, serializer: ActivityPub::CollectionSerializer
+
+  def id
+    ActivityPub::TagManager.instance.uri_for(object)
+  end
+
+  def attributed_to
+    ActivityPub::TagManager.instance.uri_for(object.parent_account)
+  end
+
+  def type
+    'Collection'
+  end
+
+  def first
+    conversation_statuses = object.statuses[0..5]
```

Comment:
According to my testing, `[]` does not alias to any special ActiveRecord query modifications, so what seems to happen here is *all* the statuses in the conversation are loaded from the database, stored in an array, then that array is sliced to the first 6 items. Perhaps data loading could be moved out of the serializer to the controller instead? The data modification logic could be moved to a presenter, then this serializer could be very simple.